Realm/RLMSyncUser.h

```diff
@@ -113,8 +140,118 @@ NS_SWIFT_UNAVAILABLE("Use the full version of this API.");
+
+ If no URL is passed in, the permission changes will be applied to ALL
+ synced Realms managed by this user.
+
+ If no user is passed in, the permission changes will be applied with
+ respect to the given synced Realm (or all synced Realms) for ALL users.
+
+ @param realmURL        The URL of the synced Realm on the server whose
+                        permissions should be modified, or nil.
+ @param identity        The identity of the user whose permissions for
+                        the Realm at `realmURL` should be modified, or
+                        `nil`.
+ @param permissions     The new permissions to be set.
+ @param callback        An optional block through which the progress of
+                        the permission change operation can be reported.
+ @returns   If `callback` was specified, a notification token. Hold onto
+            the token for as long as callback status updates are desired.
+            Call `-stop` on the token before destroying it.
+ */
+- (nullable RLMNotificationToken *)modifyRealmPermissionsForURL:(nullable NSURL *)realmURL
+                                                   userIdentity:(nullable NSString *)identity
+                                                    permissions:(RLMSyncRealmPermission)permissions
+                                                       callback:(nullable RLMSyncPermissionChangeBlock)callback
+NS_REFINED_FOR_SWIFT;
+
+/**
+ Create a permission offer.
+
+ A permission offer is repesented by a portable string token which can be
+ given to another user through any convenient means (such as email). It
+ represents offering certain permissions to the recepient, allowing them
+ to work with a synced Realm managed by this user.
+
+ The token itself is returned asynchronously through the `callback` block.
+ It is this token string which can be sent to a different user and then
+ passed into `-acceptOffer:callback:` to accept the offer. Do not confuse
+ this token with the identifier returned immediately by this method.
+
+ @param realmURL        The URL of the synced Realm on the server.
+ @param date            The date when the offer should expire. If nil, the
+                        offer will never expire.
+ @param permissions     The permissions that the offer should include.
+ @param token           A pointer to a notification token object, through
+                        which a notification token should be passed out.
+ @param callback        A block through which the status of the permission
+                        offer creation process should be reported.
+ @return    An identifier for the newly-created permission offer. This
+            identifier can be passed into any method taking an `offerKey`
+            parameter. Do not confuse this with the permission offer's
+            token, which is passed back asynchronously through the callback
+            once the server has processed the request.
+ */
+- (NSString *)offerPermissionsForURL:(NSURL *)realmURL
```

Comment:
Alternatively, why not just pass the object/token in the callback? This would mirror the `modify` API in the sense it isn't really designed for offline-use because while underneath it is creating a `PermissionChange` object, unless the callback fires, you would have no idea that object exists.